collector/__init__.py

```diff
@@ -0,0 +1,590 @@
+        
+        for sem in self.semesters:
+            tafile = 'tas_' + ssite + '_' + sem + '.txt'
+        if not os.path.exists(plandir + tafile):
+            get_report(ssite, tafile, plandir)
+            
+        tmp = get_tas(plandir + tafile)
+        if tas:
+            tas = vstack([tas, tmp])
+        else:
+            tas = tmp.copy()
+
+        return sumtas_date(tas, tadate)
+
+    def _elevation_contrains(self, elevation_type, max_elevation, min_elevation):
+        """ Calculate elevation constrains """
+
+        if elevation_type == 'NONE' or elevation_type == 'NULL':
+            elevation_type = 'AIRMASS'
+        if min_elevation == 'NULL' or min_elevation == '0.0':
+            if elevation_type == 'AIRMASS':
+                min_elevation = '1.0'
+            else:
+                min_elevation = '-5.0'
+        if max_elevation == 'NULL' or max_elevation == '0.0':
+            if elevation_type == 'AIRMASS':
+                max_elevation = MAX_AIRMASS
+            else:
+                max_elevation = '5.0'
+        self.elevation.append({'type': elevation_type, 'min': float(min_elevation), 'max': float(max_elevation)})
+
+    def _instrument_setup(self, configuration: Dict[str, List[str]], instrument_name: str) -> Instrument:
+        """ Setup instrument configurations for each observation """
+        instconfig = {} 
+
+        fpuwidths = []
+        disperser = 'NONE'
+        for key in configuration.keys():
+            ulist = uniquelist(configuration[key])
+            if key in ['filter', 'disperser']:
+                for kk in range(len(ulist)):
+                    ifnd = ulist[kk].find('_')
+                    if ifnd == -1:
+                        ifnd = len(ulist[kk])
+                ulist[kk] = ulist[kk][0:ifnd]
+            # Use human-readable slit names
+            if 'GMOS' in instrument_name:
+                if key == 'fpu':
+                    fpulist = FpuXmlTranslator(ulist)
+                    fpunames = []
+                    for fpu in fpulist:
+                        fpunames.append(fpu['name'])
+                        fpuwidths.append(fpu['width'])
+                    ulist = fpunames
+                    instconfig['fpuWidth'] = fpuwidths
+                if key == 'customSlitWidth':
+                    for cwidth in ulist:
+                        fpuwidths.append(CustomMaskWidth(cwidth))
+                    instconfig['fpuWidth'] = fpuwidths
+            if key == 'disperser':
+                disperser = ulist[0]
+            else:
+                instconfig[key] = ulist
+
+        #             print(instconfig)
+        if any(inst in instrument_name.upper() for inst in ['IGRINS', 'MAROON-X']):
+            disperser = 'XD'
+       
+        return Instrument(instrument_name,disperser,instconfig)
+        
+    def _readzip(self, 
+                 zipfile: str, 
+                 xmlselect: List[str], 
+                 site: str,
```

Comment:
Some functions only use the string version to either find a file or find something in the XML, so at some point we'll need to use it. The old code only uses str so if we implement the Provider it would be a nice wrap up between both things, I think.